### ISSUE TITLE
关于训练P_Net20时用到的激活函数SoftmxActivation参数"mode"的选择

### ISSUE THREAD

**Issue by TMaysGGS** [zuoqing1988/train-mtcnn#32]
左博士你好，

非常感谢你开源的代码。我已用你的训练方法成功训练了一个自己修改了结构的单独预测人脸关键点的模型。在尝试复现你的PNet训练代码时，收到一个关于激活函数输入维度不对应的报错，去查了mx.symbol.SoftmaxActivation的源码里的注释：
"If `mode` = ``channel``, this operator will compute a k-class softmax at each position
    of each instance, where `k` = ``num_channel``. This mode can only be used when the input array
    has at least 3 dimensions."，
显示我输入的数据维度是2，再去查看了一下输入标签（PNet训练时会输入0, -1, 1三种），我想应该是mode选择的问题，因为这里标签数据应该是(batch_size, 2)的形状。之后我将mode改成"instance"，报错消失训练顺利进行了，但能否训练成功结果还没看到。因为我是初学MXNet框架，所以想向左博士求证下想法，尤其是这里用到了自定义的Data Iterator的情况下，我也不太确定。

谢谢。
